core/src/main/java/org/elasticsearch/common/io/stream/NamedWriteableRegistry.java

```diff
@@ -31,54 +31,63 @@
     private final Map<Class<?>, InnerRegistry<?>> registry = new HashMap<>();
 
     /**
-     * Registers a {@link NamedWriteable} prototype given its category
+     * Register a {@link NamedWriteable} given its category, its name, and a function to read it from the stream.
```

Comment:
does this method still register a NamedWriteable? or do we have to update javadocs?